api4/user.go

```diff
@@ -429,6 +434,123 @@ func setDefaultProfileImage(c *Context, w http.ResponseWriter, r *http.Request)
 	ReturnStatusOK(w)
 }
 
+func getIconImage(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	canSee, err := c.App.UserCanSeeOtherUser(c.App.Session.UserId, c.Params.UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if !canSee {
+		c.SetPermissionError(model.PERMISSION_VIEW_MEMBERS)
+		return
+	}
+
+	user, err := c.App.GetUser(c.Params.UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	etag := strconv.FormatInt(user.LastPictureUpdate, 10)
+	if c.HandleEtag(etag, "Get Icon Image", w, r) {
+		return
+	}
+
+	img, readFailed, err := c.App.GetIconImage(user)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if readFailed {
+		w.Header().Set("Cache-Control", fmt.Sprintf("max-age=%v, public", 5*60)) // 5 mins
+	} else {
+		w.Header().Set("Cache-Control", fmt.Sprintf("max-age=%v, public", 24*60*60)) // 24 hrs
+		w.Header().Set(model.HEADER_ETAG_SERVER, etag)
+	}
+
+	w.Header().Set("Content-Type", "image/svg+xml")
+	w.Write(img)
+}
+
+func setIconImage(c *Context, w http.ResponseWriter, r *http.Request) {
+	defer io.Copy(ioutil.Discard, r.Body)
+
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.SessionHasPermissionToUser(c.App.Session, c.Params.UserId) {
```

Comment:
This permission is because we always expect to modify other users? Can a bot call the API if so, can it set the icon of other bot (if he has the permissions to edit other users)?

Reply:
ahh good catch. Using `c.App.SessionHasPermissionToManageBot(c.App.Session, botUserId)` instead for `set` and `delete` bot icon routes.